src/services/store.ts

```diff
@@ -231,3 +234,16 @@ export function getLogsVisualizationType(): LogsVisualizationType {
 export function setLogsVisualizationType(type: string) {
   localStorage.setItem(VISUALIZATION_TYPE_LOCALSTORAGE_KEY, type);
 }
+
+// Panel options
+const PANEL_OPTIONS_LOCALSTORAGE_KEY = `${pluginJson.id}.panel.option`;
+export interface PanelOptions {
+  panelType: AvgFieldPanelType;
+}
+export function getPanelOption<K extends keyof PanelOptions, V extends PanelOptions[K]>(option: K): V | null {
+  return localStorage.getItem(`${PANEL_OPTIONS_LOCALSTORAGE_KEY}.${option}`) as V | null;
```

Comment:
This could probably use integration with narrowing.ts

Reply:
Good catch, I removed the assertion and updated the getPanelOption to require providing all possible values for the panel options, might not work for options with many/infinite possible values, but provides better safety for these enum options.